contrib/babelfishpg_tsql/src/pltsql_utils.c

```diff
@@ -1975,15 +1977,194 @@ check_fulltext_exist(const char *schema_name, const char *table_name)
 	relid = get_relname_relid((const char *) table_name, schemaOid);
 
 
-	// Check if table exists
+	/* Check if table exists */
 	if (!OidIsValid(relid))
 		ereport(ERROR,
 			(errcode(ERRCODE_UNDEFINED_TABLE),
 				errmsg("relation \"%s\" does not exist",
 					table_name)));
 	
 	ft_index_name = get_fulltext_index_name(relid, table_name);
-	return ft_index_name != NULL;
+	if(ft_index_name == NULL)
+		return false;
+
+	ft_column_name_list = get_fulltext_indexed_columns(relid, ft_index_name);
+	/* A safety check to handle the case when the index definition changes from the current definition */
+	if(ft_column_name_list == NIL)
+	{
+		pfree(ft_index_name);
+		return false;
+	}
+
+	/* Check if column exists */
+	foreach(column_name, column_name_list)
+ 	{
+		if(!check_column_list(relid, (char *)(column_name)->ptr_value))
+		{
+			ereport(ERROR,
+				(errcode(ERRCODE_INVALID_OBJECT_DEFINITION),
+					errmsg("column \"%s\" does not exist.",
+						(char *)(column_name)->ptr_value)));
+		}
+	}
+
+	/* Check if column is fulltext indexed */
+	foreach(column_name, column_name_list)
+	{
+		bool flag = false;
+		foreach(ft_column_name, ft_column_name_list)
+		{
+			if(pg_strcasecmp((char *)(column_name)->ptr_value, (char *)(ft_column_name)->ptr_value) == 0)
+			{
+				flag = true;
+				break;
+			}
+		}
+		if(!flag)
+		{
+			list_free(ft_column_name_list);
+			ereport(ERROR,
+				(errcode(ERRCODE_INVALID_COLUMN_REFERENCE),
+					errmsg("Cannot use a CONTAINS or FREETEXT predicate on column \'%s\' because it is not full-text indexed.",
+						(char *)(column_name)->ptr_value)));
+		}
+	}
+	pfree(ft_index_name);
+	list_free(ft_column_name_list);
+	return true;
+}
+
+ 
+/* 
+ * Check if the columns provided in the freetext predicate query exist or not
+ * The check for the columns existence is handled in BISON, but we require to 
```

Comment:
Fix: Update the comment. Bison only parses the query, the actual check for column existence is done during execution time from PG side.